main.py

```diff
@@ -1,269 +1,36 @@
-        print(
-            f'T-Statistic: {self.ab_t_stat:.3f} | P-value: {self.ab_p_value:.3f} | Degrees of freedom: {self.ab_df:.3f}')
-        print('\nSample size:')
-        print(f'Control: {control_size} | Treatment: {treatment_size}')
-        print('Group Statistical data:')
-        print(f'Control: Total sales is {control_sales:.3f}  | Average sales is {control_avg_sales:.3f}')
-        print(f'Treatment: Total sales is {treatment_sales:.3f}  | Average sales is {treatment_avg_sales:.3f}')
-        print('Differences:')
-        print(f'Absolute: {absolute_diff:.3f} | Relative (lift): {relative_lift:.3f}%')
-        print('\nConfidence Interval:')
-        print(f'Absolute Difference CI: ({self.lb:.3f}, {self.ub:.3f})')
-        print(f'Relative Difference (lift) CI: ({lower_lift * 100:.3f}%, {upper_lift * 100:.3f}%)')
-        print('----------------------------------------------------------------------------------------------')
-
+# List all the files which are needed in the test
+paths = ['historical_daily_sales.csv', 'historical_daily_users.csv', 'pretest_data.csv', 'abtest_result.csv']
+file_info = FilesInfo(files_paths=paths)
 
+# List all the file paths
 historical_daily_sales_path = 'historical_daily_sales.csv'
 historical_daily_users_path = 'historical_daily_users.csv'
 pretest_data_path = 'pretest_data.csv'
 abtest_data_path = 'abtest_result.csv'
-daily_sales_data = DataReader(historical_daily_sales_path).read_csv()
-daily_users_data = DataReader(historical_daily_users_path).read_csv()
-pretest_data = DataReader(pretest_data_path).read_csv()
-abtest_data = DataReader(abtest_data_path).read_csv()
-PowerAnalysis(daily_sales_data).print_analysis_result()
-sample_size = PowerAnalysis(daily_sales_data).calculate_sample_size()
-TestSetup(sample_size, daily_users_data).print_analysis_result()
-ValidityChecks(pretest_data, abtest_data).print_analysis_result()
-ValidityChecks(pretest_data, abtest_data).aa_test_plot()
-ConductStatisticalInference(abtest_data).print_analysis_result()
+
+# Read all the csv files
+daily_sales_data = DataReader(path=historical_daily_sales_path, files_info=file_info).read_csv()
+daily_users_data = DataReader(path=historical_daily_users_path, files_info=file_info).read_csv()
+pretest_data = DataReader(path=pretest_data_path, files_info=file_info).read_csv()
+abtest_data = DataReader(path=abtest_data_path, files_info=file_info).read_csv()
+
+# Instances of the class
+power_analysis = PowerAnalysis(data=daily_sales_data)
+sample_size = power_analysis.calculate_sample_size()
+ab_test_setup = ABTestSetup(sample_size=sample_size, data=daily_users_data)
+validity_checks = ValidityChecks(aa_data=pretest_data, ab_data=abtest_data)
+stats_inference = ConductStatisticalInference(ab_data=abtest_data)
+
+# Print analysis results
+power_analysis.print_analysis_result()
+ab_test_setup.print_analysis_result()
+validity_checks.print_analysis_result()
+validity_checks.aa_test_plot()
+stats_inference.print_analysis_result()
```

Comment:
Put all the code in main in a main function, and execute the main function in a `if __name__ == "__main__"` block
https://realpython.com/if-name-main-python/